packages/reporter/src/reporters/gitlab/api/GitLabCodeQualityFileWriter.spec.ts

```diff
@@ -0,0 +1,92 @@
+import 'reflect-metadata';
+import { GitLabCodeQualityFileWriter } from './GitLabCodeQualityFileWriter';
+import type { CodeQualityReport } from '../types';
+import type { GitLabConfig } from './GitLabConfig';
+
+// Mock the fs module
+jest.mock('node:fs/promises', () => ({
+  writeFile: jest.fn()
+}));
```

Comment:
To avoid repeating `const { writeFile } = await import('node:fs/promises');` consider this approach:

```suggestion
const writeFile = jest.fn()
jest.mock('node:fs/promises', () => ({
  writeFile
}));
```

Reply:
The code you suggested doesn't work, but the idea is good, so I've tried to implement something similar. Check it, please, maybe it is not optimal.

Reply:
So, this implementation works fine. You could also avoid type casting like this:

```ts
import { writeFile } from 'node:fs/promises';

// ...

jest.mock('node:fs/promises', () => ({ writeFile: jest.fn() })); 

// ...

const mockedWriteFile = jest.mocked(writeFile);
```